src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+
+        // Alice consumes sequences (# of txns)
+        BEAST_EXPECT(env.seq(alice) == seq + 3);
+
+        // Alice pays XRP & Fee; Bob receives XRP
+        BEAST_EXPECT(env.balance(alice) == preAlice - XRP(1) - batchFee);
+        BEAST_EXPECT(env.balance(bob) == preBob + XRP(1));
+    }
+
+    void
+    testAccountDelete(FeatureBitset features)
+    {
+        testcase("account delete");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        env.fund(XRP(1000), alice, bob);
+        env.close();
+
+        // Close enough ledgers to delete account
+        int const delta = [&]() -> int {
+            if (env.seq(alice) + 300 > env.current()->seq())
+                return env.seq(alice) - env.current()->seq() + 300;
+            return 0;
+        }();
+        for (int i = 0; i < delta; ++i)
+            env.close();
+
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+
+        auto const seq = env.seq(alice);
+        auto const batchFee =
+            batch::calcBatchFee(env, 0, 2) + env.current()->fees().increment;
+        env(batch::outer(alice, seq, batchFee, tfIndependent),
+            batch::inner(pay(alice, bob, XRP(1)), seq + 1),
+            batch::inner(acctdelete(alice, bob), seq + 2),
+            batch::inner(pay(alice, bob, XRP(2)), seq + 3));
+        auto const txIDs = env.tx()->getBatchTransactionIDs();
+        TxID const parentBatchId = env.tx()->getTransactionID();
+        std::vector<TestBatchData> testCases = {
+            {"tesSUCCESS", to_string(txIDs[0])},
+            {"tesSUCCESS", to_string(txIDs[1])},
+        };
+        env.close();
+        validateBatch(env, parentBatchId, testCases);
+
+        // Alice does not exist; Bob receives Alice's XRP
+        BEAST_EXPECT(!env.le(keylet::account(alice)));
+        BEAST_EXPECT(env.balance(bob) == preBob + (preAlice - batchFee));
+    }
+
+    static uint256
+    getCheckIndex(AccountID const& account, std::uint32_t uSequence)
+    {
+        return keylet::check(account, uSequence).key;
+    }
+
+    void
+    testObjectCreateSequence(FeatureBitset features)
+    {
+        testcase("object create w/ sequence");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, gw);
+        env.close();
+
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        auto const aliceSeq = env.seq(alice);
```

Comment:
Might be worth adding a `tfIndependent` test where the CheckCreate fails and the CheckCash would have succeeded if it didn't.